src/index.js

```diff
@@ -280,7 +280,10 @@ function getMain({ options, entry, format }) {
 		mainNoExtension,
 	);
 	mainsByFormat.modern = replaceName(
-		(pkg.syntax && pkg.syntax.esmodules) || pkg.esmodule || 'x.modern.js',
+		(typeof pkg.exports === 'string' && pkg.exports) ||
+			(pkg.syntax && pkg.syntax.esmodules) ||
```

Comment:
I think this proposal was superceded by pkg.exports and can be removed

Reply:
Is anyone using that? I've never seen it in the wild, but it potentially would be a breaking change, no?

Reply:
Ah, I didn't notice it was previously supported here.